source/portable/NetworkInterface/STM32Hxx/NetworkInterface.c

```diff
@@ -801,11 +803,20 @@ static int32_t ETH_PHY_IO_WriteReg( uint32_t ulDevAddr,
 /*******************************************************************************
 *                   Ethernet Handling Functions
 *******************************************************************************/
-
-void ETH_IRQHandler( void )
-{
-    HAL_ETH_IRQHandler( &( xEthHandle ) );
-}
+#if 0
```

Comment:
I'm not very clear on the benefits of using the CubeIDE generated code for Ethernet driver and its initialization in this case. We are always defaulting to the ethernet driver present in the `source/portable/NetworkInterface/STM32Hxx` folder, even if the CubeIDE generates a different version. Also, there are couple of things that user needs to manually update in the project as you have listed in the `readme.md` to make it work with the autogenerated code, which I think defeats the purpose of the CubeIDE tool. 

I believe there are a lot of users that doesn't rely on CubeIDE generated code (atleast for ethernet peripheral), removing `ETH_IRQHandler` from the `NetworkInterface.c` will break their code, unless they manually update it.  

The only benefit that I see we get from the CubeIDE generated code are the two auto generated functions: `HAL_ETH_MspInit()` and `HAL_ETH_MspDeInit()`, which can be helpful in initializing/deinitializing the hardware (especially for the official ST development boards supported by the Cube IDE), but, again has to be manually tweaked for custom boards that runs on STM32Hxx. I think users can always refer the autogenerated version of these functions and implement their own version part of the application itself.

Reply:
@miguelfreitas, yes, I usually disable code generation for ethernet from the Cube IDE and rely on the NetworkInterface.c. While disabling Cube IDE generated code for ethernet we have to manually define (uncomment) `HAL_ETH_MODULE_ENABLED` in `Core\Inc\stm32h7xx_hal_conf.h`. 

STM32Hxx network interface was not fully updated to support the IPv6 related functionalities which got recently added to the main branch. The PRs: #970 and #965 (970 is an alternative approach for changes made in 965) should fix that. While updating those changes some of the build related fixes that you had made in your PR are already added in those PRs, which was required while testing. Will you be interested in updating your PR with respect to the latest changes or prefer us to do it on your behalf.

Reply:
Hi @tony-josi-aws, thanks for this new info. I'd prefer you to to proceed with updating this or the other PR with the latest changes as you prefer since I'll be away on vacations for the next two weeks. Thanks!

Reply:
Sure @miguelfreitas, I will update the PR.

Reply:
@miguelfreitas, merging this PR. Thanks for contibuting to FreeRTOS+TCP.